share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```
@@ -384,5 +384,70 @@ function(sysbuild_module_call)
       endforeach()
     endif()
   endforeach()
+endfunction()
+
+# Usage:
+#   sysbuild_cache_set(VAR <variable> [APPEND [REMOVE_DUPLICATES]] <value>)
+#
+# This function will set the specified value of the sysbuild cache variable in
+# the CMakeCache.txt file which can then be accessed by images.
+# `VAR` specifies the variable name to set/update.
+#
+# The result will be returned in `<variable>`.
+#
+# Example use:
+#   sysbuild_cache_set(VAR ATTRIBUTES APPEND REMOVE_DUPLICATES battery)
+#     Will add the item `battery` to the `ATTRIBUTES` variable as a new element
+#     in the list in the CMakeCache and remove any duplicates from the list.
+#
+# <variable>:        Name of variable in CMake cache.
+# APPEND:            If specified then will append the supplied data to the
+#                    existing value as a list.
+# REMOVE_DUPLICATES: If specified then remove duplicate entries contained
+#                    within the list before saving to the cache.
+# <value>:           Value to set/update.
+function(sysbuild_cache_set)
+  cmake_parse_arguments(VARS "APPEND;REMOVE_DUPLICATES" "VAR" "" ${ARGN})
```

I believe quoting would be recommended here in if we think it is important to support the `"\;"` use-case. 

```suggestion
  cmake_parse_arguments(VARS "APPEND;REMOVE_DUPLICATES" "VAR" "" "${ARGN}")
```

---

Looked through all the files in `cmake` in zephyr and none of them quote this parameter